File: Sort/insertionSort.go
package main

import "fmt"

func insertionSort(items []int) {
	n := len(items)
	for i := 1; i < n; i++ {

		j := i
		for j > 0 {
			if items[j-1] > items[j] {
				items[j-1], items[j] = items[j], items[j-1]
			}
			j = j - 1

		}

	}
}

func main() {
	items := []int{1, 2, 6, 8, 5, 4, 3, 7}
	fmt.Println(items)
	insertionSort(items)
	fmt.Println(items)

}

File: Sort/insertionSort.py
def insertionsort(a):
    n=len(a)
    for i in range(1,n):
        cvalue=a[i]
        position=i 
        while position>0 and a[position-1]>cvalue:
            a[position]=a[position-1]
            position=position-1 
        a[position]=cvalue
    


A = [3,5,8,9,6,2]
print('Original Array:',A)
insertionsort(A)
print('Sorted Array:',A)
